plutus-scb/src/Cardano/Node/Server.hs

```diff
@@ -48,27 +50,33 @@ main MockServerConfig { mscBaseUrl
                       , mscInitialTxWallets
                       } availability =
     runStdoutLoggingT $ do
-        stateVar <- liftIO $ newMVar (initialAppState mscInitialTxWallets)
+        blockQueue <- liftIO $ newTQueueIO
+        txQueue    <- liftIO $ newTQueueIO
+        -- Local chain state (node client)
+        clientStateVar <- liftIO $ newMVar (initialAppState mscInitialTxWallets)
+        -- Global chain state (node server)
+        serverStateVar <- liftIO $ newMVar (initialAppState mscInitialTxWallets)
```

Comment:
And how does the `clientStateVar` stay in sync with the `serverStateVar` (I'm assuming via `handleChain` which sends the transactions to the server?)

Reply:
I have redefined the `Chain` and `Follower` effect handlers to achieve synchronisation. 

Whenever I `queueTx` I also send the transaction to the transaction queue (which is forwarded using node protocols to the server) and then update the local state.

And when a node follower asks for new blocks, I flush the blocks Queue (which contains blocks received from the server using the node protocols) and update the local state. Then I use the normal flow that queries the local state for new blocks.

Reply:
Okay. I think in the future there will be some work in separating out the bits of the PAB that are only intended to be used in testing/simulation mode but this is fine for now.